gatsby-node.js

```diff
@@ -194,60 +84,179 @@ async function doQuery({
     );
   }
 
-  setQueryStatus(`graphql resulted in ${objects.length} records`);
+  // return a map by id for later use
+  return objects.reduce((acc, object = {}) => {
+    return {
+      ...acc,
+      [object.objectID]: object,
+    };
+  }, {});
+}
 
-  let hasChanged = objects;
-  if (enablePartialUpdates) {
-    setQueryStatus(`Starting Partial updates...`);
+// get all match fields for all queries to minimize calls to the api
+function getAllMatchFields(queries, mainMatchFields = []) {
+  const allMatchFields = [...mainMatchFields];
 
-    const algoliaObjects = await fetchAlgoliaObjects(indexToUse, matchFields);
+  queries.forEach(({ matchFields = [] }) => {
+    matchFields.forEach(field => {
+      if (!allMatchFields.includes(field)) {
```

Comment:
this could be done by making allMatchFields a set and spreading it, which is pretty much the same, but consistent with how it's done in replicas now

Reply:
Sounds good to me, will fix!